### DISCUSSION TITLE
Strong text between strong+emphasis text

### DISCUSSION THREAD

**Discussion by chartinger** [unifiedjs/unified#160] [Q&A]
I am currently doing experiments with [Milkdown](https://milkdown.dev/#/) which uses remark in the background. When saving strong text inside a string and emphasis text, the resulting markdown cannot be parsed again the same way.

Using the output from [Pandoc](https://pandoc.org/try/) the resulting html does look different

Testcase:
```
  t.equal(
    micromark('***foo****bar****baz***'),
    '<p><strong><em>foo</em></strong><em>bar</em><strong><em>baz</em></strong></p>',
    'should support strong inside strong and emphasis'
  )
```
Output:
```
    expected: |-
      '<p><strong><em>foo</em></strong><em>bar</em><strong><em>baz</em></strong></p>'
    actual: |-
      '<p><em><strong>foo</strong>**bar</em><em><strong>baz</strong></em></p>'
```

So my question is: Is this a bug in the parser or is the markdown not valid according to the specs?

**Comment by wooorm** [comment]
We try to adhere to CommonMark. So pasting this input there: <https://spec.commonmark.org/dingus/?text=***foo****bar****baz***>. Seems that they yield something different then what micromark does (but it’s not your expected value either).

**Reply by wooorm** [reply]
See also: <https://babelmark.github.io/?text=***foo****bar****baz***>.

**Reply by chartinger** [reply]
Thanks, seems the correct test case for micromark according to commonmark should be:

```
  t.equal(
    micromark('***foo****bar****baz***'),
    '<p><em><strong>foo</strong></em><em>bar</em><em><strong>baz</strong></em></p>',
    'should support strong inside strong and emphasis'
  )
```
```
    expected: |-
      '<p><em><strong>foo</strong></em><em>bar</em><em><strong>baz</strong></em></p>'
    actual: |-
      '<p><em><strong>foo</strong>**bar</em><em><strong>baz</strong></em></p>'
```

**Reply by wooorm** [reply]
This might be related/the same as https://github.com/syntax-tree/mdast-util-from-markdown/issues/15

**Reply by chartinger** [reply]
After a bit more investigation i wonder if equal parsing to commonmark is even possible with the current micromark parser strategy.

Extending the attention skip code with a special case for nesting parsed sections "solves" my testcase and the one from the linked issue
```
          const leftLength = events[open][1].end.offset - events[open][1].start.offset;
          const rightLength = events[index][1].end.offset - events[index][1].start.offset;
          const totalLength = leftLength + rightLength;

...

              (rightLength) % 3 && !(totalLength % 3) && (
                ((leftLength > rightLength) && ((events[open+1][1].type !== 'emphasis'))) ||
                ((leftLength < rightLength) && ((events[open+1][1].type !== 'strong')))
              )
```
but still fails with more crazy strings like `*******foo****sdf*****var*****` Without checking the code, it seems CommonMark prioritizes by *-count somehow compared to the location-based parsing from micromark.
